src/Illuminate/Foundation/Testing/Concerns/InteractsWithGate.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace Illuminate\Foundation\Testing\Concerns;
+
+use Illuminate\Foundation\Testing\ExpectedPermission;
+use Illuminate\Support\Facades\Gate;
+
+trait InteractsWithGate
+{
+    /**
+     * Indicates if the hooks have been registered.
+     *
+     * @var bool $hasRegisteredGateHooks
+     */
+    private $hasRegisteredGateHooks = false;
```

Comment:
Please change all `private` visibilities to `protected`.

Reply:
@driesvints I changed most visibilities to `protected` but this is intentionally left `private` because only `shouldAllow` and `shouldDeny` should be expose to the consumers of the trait.